src/data/polynomial/ring_division.lean

```diff
@@ -422,6 +426,48 @@ calc ((roots ((X : polynomial R) ^ n - C a)).card : with_bot ℕ)
       ≤ degree ((X : polynomial R) ^ n - C a) : card_roots (X_pow_sub_C_ne_zero hn a)
   ... = n : degree_X_pow_sub_C hn a
 
+lemma le_root_multiplicity_map {K L : Type*} [comm_ring K] [is_domain K]
+  [comm_ring L] [is_domain L] {p : polynomial K} {f : K →+* L} (hf : function.injective f)
```

Comment:
I different version might assume `map f p ≠ 0` instead of `function.injective f`.

Reply:
Shall we add a ' version then (or of map_ne_zero)?